smart-contracts/test/helpers/deployLocks.js

```diff
@@ -1,21 +1,37 @@
 const PublicLock = artifacts.require('./PublicLock.sol')
 const Locks = require('../fixtures/locks')
 
-module.exports = function deployLocks (unlock) {
+module.exports = function deployLocks (unlock, from) {
   let locks = {}
-
   return Promise.all(
-    Object.keys(Locks).map(name => {
-      return unlock.createLock(
-        Locks[name].expirationDuration.toFixed(),
-        Locks[name].keyPrice.toFixed(),
-        Locks[name].maxNumberOfKeys.toFixed()
-      ).then(async (tx) => {
+    Object.keys(Locks).map(async name => {
+      let createCall
+      if (unlock.methods) {
+        createCall = unlock.methods.createLock(
+          Locks[name].expirationDuration.toFixed(),
+          Locks[name].keyPrice.toFixed(),
+          Locks[name].maxNumberOfKeys.toFixed()
+        )
+      } else {
```

Comment:
else this is used if there's no proxy... Is that the gist of the change here?

Reply:
> else this is used if there's no proxy... Is that the gist of the change here?

Yes - although not specifically for the proxy.  It's about supporting both the web3 API and the truffle API.  Now this helper should just work given either.  I suspect we will be migrating more to the new syntax in the future.

Reply:
Got it. looks good.